app/create_file.py

```diff
@@ -1 +1,56 @@
-# write your code here
+from datetime import datetime
+from sys import argv
+from os import mkdir, chdir
+
+filename = ""
+directories = []
+
+if "-f" in argv:
+    index = 0
+
+    for _ in range(len(argv) - 1):
+        index += 1
+        if argv[_: _ + 1] == ["-f"]:
+            break
+
+    filename = argv[index]
+
+    if "-d" in argv:
+        index2 = 0
+
+        for _ in range(len(argv) - 1):
+            index2 += 1
+            if argv[_: _ + 1] == ["-d"]:
+                break
+
+        directories = argv[index2:index - 1]
```

Comment:
You can do almost the same thing to capture directories, just add some specific logic to identify when the directory names end and the file name flag begins. To do this, you need to:
- [ ] Assign the index `-d` to a variable (let's call it `d_index`)
- [ ] Check if the `-f` flag exists and if it is after `d_index`, comparing which index is greater
- [ ] If the `-f` index is greater, assign its index to a variable `end_index` (which will be where the directories will end)
- [ ] If `d_index` is greater, all the other elements of `argv` after `d_index` will be directory names, so you assign the length of `argv` to the variable `end_index`
- [ ] The directories will be a slice of the list `argv`, starting at the next element of `d_index` and ending at `end_index`

In the end, your code will look like this:

```suggestion
    if "-d" in argv:
        d_index = argv.index("-d")

        if "-f" in argv and argv.index("-f") > d_index:
            end_index = argv.index("-f")
        else:
            end_index = len(argv)

        directories = argv[d_index + 1:end_index]
```